frontend/src/modules/scaffold/views/scaffoldAssetItemView.js

```diff
@@ -0,0 +1,211 @@
+define([
+  'core/origin',
+  'backbone-forms',
+  'core/helpers',
+  './scaffoldAssetView',
+  'core/models/courseAssetModel',
+  'modules/assetManagement/views/assetManagementModalView',
+  'modules/assetManagement/collections/assetCollection'
+], function(Origin, BackboneForms, Helpers, ScaffoldAssetView, CourseAssetModel, AssetManagementModalView, AssetCollection) {
+
+  var ScaffoldAssetItemView = ScaffoldAssetView.extend({
+
+    events: function() {
+      return _.extend({}, ScaffoldAssetView.prototype.events, {
+        'click .add-item': 'onAddItemClick',
+        'click .ui-draggable': 'onDragItemClick'
+      });
+    },
+
+  	onAddItemClick: function(event) {
+      event.preventDefault();
+      this.itemListView.addItem();
+    },
+
+    onDragItemClick: function(event) {
+      var item = _.find(this.dragItems, function(i) {
+        return i.elm[0] === event.currentTarget;
+      });
+      item.view.openEditor();
+    },
+
+    initialize: function(options) {
+      ScaffoldAssetView.prototype.initialize.apply(this, arguments);
+      this.init(options);
+    },
+
+    init: function(options) {
+      this.dragItems = [];
+      this.precision = options.schema.editorAttrs['data-precision'] || 3;
```

Comment:
We don't need to shoehorn attributes into Backbone Form's `editorAttrs` – that's designed for attributes to be used in the DOM.

Instead, use the object form of the `inputType` attribute e.g.
```json
"inputType" : {
  "type": "AssetItem",
  "precision": 2
}
```
In the view:
```js
var attributes = options.schema.inputType;

this.precision = attributes['data-precision'] || 3;
```